### ISSUE TITLE
[Enhancement] Remove files from phone after upload

### ISSUE THREAD

**Issue by Bugsbane** [owncloud/android#601]
I mainly use ownCloud as a place to store and share photos and videos. I like to use high quality compression settings, meaning my phone uses it's storage space quickly.

Given that I already use instant upload, I would be helpful if there was an option to remove files successfully uploaded (especially by instant upload) so that I could avoid constantly running out of space and having to manually delete files that have already been uploaded to ownCloud.

**Comment by Bugsbane** [comment]
I'm not sure this is a duplicate.

As far as I can tell, #346 is about not duplicating uploaded files on the device so there aren't 2 copies on the local device. That is is definitely needed for my usecase, however this issue was about removing the *original* copy from the device after use.

Right now, uploaded files exist twice on the local device.
#346 would make sure they only existed once.
In this issue I want them to not exist at all (on the local device).

**Comment by tobiasKaminsky** [comment]
You are correct.
Re-opening it.

**Comment by tobiasKaminsky** [comment]
@Bugsbane with the current version you could just delete your instantupload folder from time to time.
Is this enough for you?
Else we could a third option to "original file will be..." called "deleted.
@jancborchardt @davivel

**Comment by Bugsbane** [comment]
I see in the beta app, there's an option that says "Original file will be..  Moved to the app folder". If " the app folder " means the ownCloud instant upload folder and moved is being used in the normal meaning of the word (ie not duplicated), and if uploads can be relied on to upload every file requested eventually, ( #919? ) then as far as I can see, my use case should already be covered without needing to manually go in and delete anything.

Maybe I'm misunderstanding something though.

**Comment by tobiasKaminsky** [comment]
This is also in current version.
And yeah, your assumptions are all correct.

**Comment by Bugsbane** [comment]
Cool. Well, sounds like we can close this then, as it seems the functionality I was asking for has been implemented...

**Comment by tobiasKaminsky** [comment]
Please try it first and confirm that it is working they way you want it :)

**Comment by Bugsbane** [comment]
Will do. I'm actually having some non oc related issues with my server at the moment that make this difficult, but when they're resolved, I'll give it a shot.

**Comment by PanicDriven** [comment]
From my understanding of this issue / request, the op (Bugsbane) is wanting an option to have the files he uploads from from his mobile app (whether it be instantupload or manual upload) to be deleted automatically once it's successfully uploaded. If this is correct, then this is an function i really want too. But i'm confused at how the original requested function has been implemented by "Original file will be.. Moved to the app folder" as this still stores the uploaded file on the local storage of the device. unless this was what Bugsbane was wanting the whole time and i misunderstood. 

I have a decent amount of users (family and friends) using my OwnCloud server. My issue is that their devices have an extremely small amount of local storage space, But they all have unlimited mobile data. (DirecTV & AT&T customers like myself) We all use OwnCloud for storage. All files uploaded are not desired to be stored on our devices. Currently i only find the option (beta and final app) to keep the file in its original location and copy it to the com.owncloud dir or to just move it to the com.owncloud dir. We currently have to delete the local data manually every time, which is very frustrating and almost makes the app unusable.

**Comment by Bugsbane** [comment]
You're completely correct @PanicDriven... I *did* mean for the images to be *completely deleted* from the local device after they had been confirmed as having been successfully uploaded to ownCloud. My usecase is exactly as the one you described for your family and friends. I don't want the files taking up space on my device, but rather be immediately uploaded and deleted locally.

Accordingly, now that I've tried the "move" option and found that it just moves it to a different folder, still on my local device, I agree that the current implementation does not meet the usecase outlined in my original post. (Sorry whoever developed it...)

**Comment by PanicDriven** [comment]
Hopefully we will be able to come to an agreement with @tobiasKaminsky about getting this feature implemented as an option. I'll do the most i can to contribute.

**Comment by davivel** [comment]
Well, that's why this issue keeps open, it still was not satisfied :) .

I also think that an option to remove after upload would make sense, specially for instant uploads.

Meanwhile, maybe you find useful next "trick"; if you enable the option "move into OC folder", later you can open the OC app and delete locally all the pictures with the action "Remove" on the parent folder containing them all, and selecting the option "Local only" in the dialog next. Not a solution, but at least you won't need to delete every file individually, and the path is easier to find that the path of the camera app in a file browser.

**Comment by stratoss** [comment]
I'm checkout out OwnCloud at the moment, and really interested is this functionality implemented already?

**Comment by davigonz** [comment]
Hi @stratoss , it is not implemented yet but contributions are always welcome 😉

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
